workflow-templates/psalm-matrix.yml

```diff
@@ -12,23 +12,37 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      ocp-branches: ${{ steps.versions.outputs.ocp-branches }}
+      php-max: ${{ steps.versions.outputs.php-max-list }}
+    steps:
+      - name: Checkout app
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      - name: Get version matrix
+        id: versions
+        uses: icewind1991/nextcloud-version-matrix@c2bf575a3516752db5ce2915499d3f694885e2c7 # v1.0.0
+
   static-analysis:
     runs-on: ubuntu-latest
+    needs: matrix
     strategy:
       # do not stop on another job's failure
       fail-fast: false
       matrix:
-        ocp-version: [ 'dev-master', 'dev-stable28', 'dev-stable27', 'dev-stable26' ]
+        ocp-version: ${{fromJson(needs.matrix.outputs.ocp-branches)}}
+        php-versions: ${{ fromJson(needs.matrix.outputs.php-max) }}
```

Comment:
Will this work, when server master allows 8.1-8.3 but stable26 only 8.0-8.2?

Especially for "matrix" apps that support a wider range this would be handy if the matching php-max would always be from the branch, otherwise we should use `php-min` ?

Reply:
Switched to a generated matrix that matches server and php versions